### ISSUE TITLE
What is your plan?

### ISSUE THREAD

**Issue by MrHinsh** [pratiksharma897/VSTS_MSID_TO_ORGID_MIGRATION#1]
How do you plan on making this transition? I dont know of any api that would allow you to do this. My understanding is that you have to contact MSFT support before you do the change and they will do some magic on their end...

**Comment by pratiksharma-dev** [comment]
Hi Martin,
I plan to proceed using the VSTS client library as I have already succeeded in Adding members from AD to VSTS. I need to find a way to Add them to teams and groups. I have already raised a case with Microsoft and it has been quite a while therefore i thought of taking help from community. Do inform me if you know anyone who can help me out.
Thanks & Regards
Pratik

**Comment by MrHinsh** [comment]
I am not sure I understand. The goal, as per the title, is to migrate from MSA to AAD, however this is not something that you can code. you have to go to MS and get them to do it. 

What part of that process are you trying to automate?

**Comment by pratiksharma-dev** [comment]
I am trying to automate the part in which i assign proper groups and permissions to Org accounts. Also removing the MSA from VSTS is in the pending tasks list.

As i am aware it is possible to add users to TFS groups programmatic ally. Also VSTS is essentially TFS 2015 in the backend. I am able to add users to VSTS groups using TFSSecurity.exe but it is not a standalone tool and sign in is still manual. Therefore i am trying to figure out something in that line...maybe by using the 
TFS client library to login to VSTS. 
Kindly refer this post: http://stackoverflow.com/questions/8018374/how-to-add-users-programmatically-to-tfs
I am trying to figure out to use this code to add users to VSTS.
This post was also of interest to me : http://geekswithblogs.net/TarunArora/archive/2011/09/30/tfs-sdk-get-groups-users-permissions-using-tfs-api-with.aspx.

Appreciate your help in this :)

**Comment by MrHinsh** [comment]
You cant. You need to raise a support call with MSFT BEFORE you switch over to AAD... then everything transitions over. They can help you fix the accounts on the back end, but only if done first.

Then you don't need to add AAD as all of the existing MSA's are converted.

**Comment by pratiksharma-dev** [comment]
Yup, will have a call with Microsoft regarding this..lets see what they have to say in this..,
